From fe5eb25385391adc4259e8081731607cf01b81fe Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:27 -0500
Subject: [drm] radeon: sync all BOs involved in a CS

Message-id: <1418852380-13061-77-git-send-email-rclark@redhat.com>
Patchwork-id: 102210
O-Subject: [RHEL7 drm 76/89] drm/radeon: sync all BOs involved in a CS v2
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.18

commit 86b276385c6a986872e4cd144f5940b156053c3f

Author:     Christian Koenig <christian.koenig@amd.com>
AuthorDate: Thu Nov 27 13:12:58 2014 +0100
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Mon Dec 15 14:27:43 2014 -0500

    drm/radeon: sync all BOs involved in a CS v2

    Not just the userspace relocs, otherwise we won't wait
    for a swapped out page tables to be swapped in again.

    v2: rebased on Alex current drm-fixes-3.18

    Signed-off-by: Christian Koenig <christian.koenig@amd.com>
    Cc: stable@vger.kernel.org
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/radeon_cs.c b/drivers/gpu/drm/radeon/radeon_cs.c
index c50a1da..4aa1ede 100644
--- a/drivers/gpu/drm/radeon/radeon_cs.c
+++ b/drivers/gpu/drm/radeon/radeon_cs.c
@@ -226,14 +226,11 @@ static int radeon_cs_get_ring(struct radeon_cs_parser *p, u32 ring, s32 priority
 
 static void radeon_cs_sync_rings(struct radeon_cs_parser *p)
 {
- int i;
-
- for (i = 0; i < p->nrelocs; i++) {
-  if (!p->relocs[i].robj)
-   continue;
+ struct radeon_cs_reloc *reloc;
 
+ list_for_each_entry(reloc, &p->validated, tv.head) {
   radeon_semaphore_sync_to(p->ib.semaphore,
-      p->relocs[i].robj->tbo.sync_obj);
+      reloc->robj->tbo.sync_obj);
  }
 }
 
-- 
1.7.1